src/test/java/org/opensearch/neuralsearch/highlight/SemanticHighlighterIT.java

```diff
@@ -0,0 +1,377 @@
+     *         "options": {
+     *             "model_id": "sentence-highlighting-model-id"
+     *         }
+     *     }
+     * }
+     */
+    public void testMatchQueryWithSemanticHighlighting() {
+        initializeTestIndex();
+        String sentenceHighlightingModelId = prepareSentenceHighlightingModel();
+
+        MatchQueryBuilder matchQuery = new MatchQueryBuilder(TEST_TEXT_FIELD_NAME, TEST_QUERY_TEXT);
+
+        Map<String, Object> searchResponse = searchWithSemanticHighlighter(
+            TEST_BASIC_INDEX_NAME,
+            matchQuery,
+            10,
+            TEST_TEXT_FIELD_NAME,
+            sentenceHighlightingModelId
+        );
+
+        verifyHighlightResults(searchResponse, "artificial intelligence");
+    }
+
+    /**
+     * Tests semantic highlighting with term query:
+     * {
+     *     "query": {
+     *         "term": {
+     *             "test-text-field": "intelligence"
+     *         }
+     *     },
+     *     "highlight": { ... }
+     * }
+     */
+    public void testTermQueryWithSemanticHighlighting() {
+        initializeTestIndex();
+        String sentenceHighlightingModelId = prepareSentenceHighlightingModel();
+
+        TermQueryBuilder termQuery = new TermQueryBuilder(TEST_TEXT_FIELD_NAME, "intelligence");
+
+        Map<String, Object> searchResponse = searchWithSemanticHighlighter(
+            TEST_BASIC_INDEX_NAME,
+            termQuery,
+            10,
+            TEST_TEXT_FIELD_NAME,
+            sentenceHighlightingModelId
+        );
+
+        verifyHighlightResults(searchResponse, "intelligence");
+    }
+
+    /**
+     * Tests semantic highlighting with boolean query:
+     * {
+     *     "query": {
+     *         "bool": {
+     *             "must": [
+     *                 { "match": { "test-text-field": "artificial" } },
+     *                 { "match": { "test-text-field": "intelligence" } }
+     *             ],
+     *             "should": [
+     *                 { "match": { "test-text-field": "learning" } }
+     *             ]
+     *         }
+     *     },
+     *     "highlight": { ... }
+     * }
+     */
+    public void testBooleanQueryWithSemanticHighlighting() {
+        initializeTestIndex();
+        String sentenceHighlightingModelId = prepareSentenceHighlightingModel();
+
+        BoolQueryBuilder boolQuery = new BoolQueryBuilder().must(new MatchQueryBuilder(TEST_TEXT_FIELD_NAME, "artificial"))
+            .must(new MatchQueryBuilder(TEST_TEXT_FIELD_NAME, "intelligence"))
+            .should(new MatchQueryBuilder(TEST_TEXT_FIELD_NAME, "learning"));
+
+        Map<String, Object> searchResponse = searchWithSemanticHighlighter(
+            TEST_BASIC_INDEX_NAME,
+            boolQuery,
+            10,
+            TEST_TEXT_FIELD_NAME,
+            sentenceHighlightingModelId
+        );
+
+        verifyHighlightResults(searchResponse, "artificial intelligence");
+    }
+
+    /**
+     * Tests semantic highlighting with query string:
```

Comment:
curious why you picked this query type, is this something users are used/planning to use often with the highlighter?

Reply:
No specific reason for this type, wanted to cover some query cases in the test.